### PR TITLE
[getdns] update to 1.7.3

### PR DESCRIPTION
**Pull request by jiayuehua**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
FTR:
~~~
D:\buildtrees\getdns\src\getdns-1-9f99f372dc.clean\src\gldns\parseutil.c(17): fatal error C1083: Cannot open include file: 'sys/time.h': No such file or directory
D:\buildtrees\getdns\src\getdns-1-9f99f372dc.clean\src\gldns\parse.c(16): fatal error C1083: Cannot open include file: 'strings.h': No such file or directory
~~~

**Comment by jiayuehua** [comment]
@dg0yt I can't see the error message in CI. Thanks for your hint.

**Comment by dg0yt** [comment]
> @dg0yt I can't see the error message in CI. Thanks for your hint.

You can.
https://github.com/microsoft/vcpkg/discussions/31357

**Comment by jiayuehua** [comment]
I know that. But likely this port error message is very simple.

File: ports/getdns/fix-include.patch [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+         { GLDNS_DIR_TTL, "$TTL" },
+diff --git a/src/gldns/parseutil.c b/src/gldns/parseutil.c
+index 293496a..bcd8e6f 100644
+--- a/src/gldns/parseutil.c
++++ b/src/gldns/parseutil.c
+@@ -14,8 +14,12 @@
+ 
+ #include "config.h"
+ #include "gldns/parseutil.h"
++#ifdef HAVE_SYS_TIME_H
```
Comment (BillyONeal) [inline ports/getdns/fix-include.patch:22]: I would like to see what happens with https://github.com/getdnsapi/getdns/pull/538 before proceeding here.

File: ports/getdns/disable-docs.patch [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
-index 413709da..d3dd03f0 100644
+index 9170be7..def3758 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -1039,9 +1039,9 @@ configure_file(src/version.c.in version.c)
+@@ -955,7 +955,7 @@ if (BUILD_TESTING)
+     message(WARNING "Test programs require libcheck and so do not build on Windows, skipping.")
+   else ()
+     find_package(Check "0.9.6" REQUIRED)
+-    set(GETDNS_TEST_SOURCE_FILES
```
Comment (BillyONeal) [inline ports/getdns/disable-docs.patch:9]: I think this should be removable from the patch

File: ports/getdns/fix-include.patch (hunk 1 above)
Comment (BillyONeal) [reply ports/getdns/fix-include.patch:22]: OK, upstream has had a couple months to respond and they have not so I think you can proceed but please evaluate just taking that patch submitted upstream.

Comment (jiayuehua) [reply ports/getdns/fix-include.patch:22]: https://github.com/getdnsapi/getdns/pull/543

File: ports/getdns/disable-docs.patch (hunk 2 above)
Comment (jiayuehua) [reply ports/getdns/disable-docs.patch]: just has removed

**Comment by MonicaLiu0311** [comment]
The following error occurred when I tested feature `libuv`:
```
-- Configuring x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" G:/getdns/buildtrees/getdns/src/getdns-1-6cf4cc7bfb.clean -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=G:/getdns/packages/getdns_x64-windows/debug -DFETCHCONTENT_FULLY_DISCONNECTED=ON -DBUILD_GETDNS_QUERY=OFF -DBUILD_GETDNS_SERVER_MON=OFF -DENABLE_STATIC=0 -DENABLE_SHARED=1 -DBUILD_LIBEVENT2=ON -DBUILD_LIBUV=ON -DBUILD_TESTING=OFF -DENABLE_STUB_ONLY=ON "-DCMAKE_MAKE_PROGRAM=C:/Program Files/Microsoft Visual Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -DBUILD_SHARED_LIBS=ON -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=G:/getdns/scripts/toolchains/windows.cmake -DVCPKG_TARGET_TRIPLET=x64-windows -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=v143 -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=G:/getdns/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON "-DVCPKG_CXX_FLAGS=/D_CRT_DECLARE_NONSTDC_NAMES " -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= "-DVCPKG_C_FLAGS=/D_CRT_DECLARE_NONSTDC_NAMES " -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=G:/getdns -D_VCPKG_INSTALLED_DIR=G:/getdns/installed -DVCPKG_MANIFEST_INSTALL=OFF
    Working Directory: G:/getdns/buildtrees/getdns/x64-windows-dbg
    Error code: 1
    See logs for more information:
      G:\getdns\buildtrees\getdns\config-x64-windows-dbg-CMakeCache.txt.log
      G:\getdns\buildtrees\getdns\config-x64-windows-dbg-out.log
      G:\getdns\buildtrees\getdns\config-x64-windows-dbg-err.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:267 (vcpkg_execute_required_process)
  ports/getdns/portfile.cmake:30 (vcpkg_cmake_configure)
  scripts/ports.cmake:147 (include)
```
**G:\getdns\buildtrees\getdns\config-x64-windows-dbg-err.log:**
```
CMake Error at CMakeLists.txt:907 (target_link_libraries):
  Target "getdns_ext_uv_shared" links to:

    Libuv::Libuv

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```

**Comment by jiayuehua** [comment]
@MonicaLiu0311  please try again, although I can't reproduce the libuv issue localy on x64-windows

**Comment by MonicaLiu0311** [comment]
All features are tested successfully in the following triplet:
```
x86-windows
x64-windows
x64-windows-static
```

**Comment by MonicaLiu0311** [comment]
If you're ready, please revert to "Ready for review".

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**